compiler/ballerina-lang/src/main/java/org/ballerinalang/model/elements/PackageID.java

```diff
@@ -133,14 +135,28 @@ public PackageID(Name orgName, List<Name> nameComps, Name version) {
                 nameComps.stream()
                         .map(Name::getValue)
                         .collect(Collectors.joining(".")));
+        // TODO: The package name should be distinguishable when the pkgName != moduleName
```

Comment:
Let's create an issue as well to track this.

Reply:
Created a task issue at: #31154